src/appengine/handlers/cron/triage.py

```diff
@@ -249,12 +252,62 @@ def _check_and_update_similar_bug(testcase, issue_tracker):
 
   return False
 
+def _get_job_bugs_filing_max(job_type):
+  """Get bugs filing max for given job.."""
+  if job_type in bug_filing_max_24_hours_per_job:
+    return bug_filing_max_24_hours_per_job.get(job_type)
+  
+  max_bugs = None
+  job = data_types.Job.query(data_types.Job.name == job_type).get()
+  if job and 'BUG_FILING_MAX_24_HOURS_PER_JOB' in job.get_environment():
+    max_bugs = int(job.get_environment().get('BUG_FILING_MAX_24_HOURS_PER_JOB'))
+  
+  bug_filing_max_24_hours_per_job[job_type] = max_bugs
+
+  return max_bugs
+
+
+def _throttle_bug(testcase, bug_filed_24_hours_per_job,
+                  bug_filed_24_hours_per_project):
+  """Return whether current bug needs to be throttled."""
+  bug_filing_job_max = _get_job_bugs_filing_max(testcase.job_type)
+  valid_timestamp = datetime.datetime.now() - datetime.timedelta(hours=24)
+  count = 0
+  if bug_filing_job_max:
+    if testcase.job_type in bug_filed_24_hours_per_job:
+      count = bug_filed_24_hours_per_job.get(testcase.job_type)
+    else:
+      count = data_types.FiledBug.query(
+          data_types.FiledBug.job_type == testcase.job_type
+          and data_types.FiledBug.timestamp >= valid_timestamp).count()
+    if count < bug_filing_job_max:
+      bug_filed_24_hours_per_job[testcase.job_type] = count + 1
+      return False
+  else:
+    if testcase.project_name in bug_filed_24_hours_per_project:
+      count = bug_filed_24_hours_per_project.get(testcase.project_name)
+    else:
+      count = data_types.FiledBug.query(
+          data_types.FiledBug.project_name == testcase.project_name
+          and data_types.FiledBug.timestamp >= valid_timestamp).count()
+    if count < issue_tracker_utils.get_issue_tracker_project_bug_filing_max():
+      bug_filed_24_hours_per_project[testcase.project_name] = count + 1
+      return False
+  return True
 
-def _file_issue(testcase, issue_tracker):
+def _file_issue(testcase, issue_tracker, bug_filed_24_hours_per_job,
+                bug_filed_24_hours_per_project):
   """File an issue for the testcase."""
   filed = False
   file_exception = None
 
+  if _throttle_bug(testcase, bug_filed_24_hours_per_job, 
+                   bug_filed_24_hours_per_project):
+    logs.log(f'Skipping bug filing for {testcase.key.id()} as it is throttled')
+    _add_triage_message(
+        testcase, 'Skipping filing as this is throttled.')
```

Comment:
I would include more specific info about the bug in the message to assist in debugging.
For example, 1) the number of bugs filed, 2) The time window, and 3) The project/job that exceeds the limit.